smbios/decoder.go

```diff
@@ -265,6 +265,15 @@ func (d *Decoder) next() (*Structure, error) {
 			}
 		}
 
+		if memInfo.TotalWidth > 0 {
+			totalWidth := uint64(memInfo.TotalWidth)

```

Comment:
You do not need a totalWidth.

Reply:
Memory Total Width value has to be extracted. Hence added a check for the same.

Reply:
Ok will remove the temporary variables

Reply:
Removed the temporary variables